server/activate.go

```diff
@@ -20,8 +22,9 @@ func (p *Plugin) OnActivate() error {
 	p.API.LogDebug("Activating NPS plugin")
 
 	if !p.canSendDiagnostics() {
-		p.API.LogDebug("Not activating NPS plugin because diagnostics are disabled")
-		return nil
+		errMsg := "Not activating NPS plugin because diagnostics are disabled"
+		p.API.LogError(errMsg)
+		return fmt.Errorf(errMsg)
```

Comment:
This can be `errors.New` to avoid the extra import
```suggestion
		return errors.New(errMsg)
```